### ISSUE TITLE
Documenting character or name strings (such as keywords) not present as variables.

### ISSUE THREAD

**Issue by YvanFournier** [doxygen/doxygen#7952]
Hello,

It would be very useful in some cases to be able to document specific character strings which do not appear directly as variables in source code, but correspond to reserved run-time file names or text-based keywords passed to functions.

I have not found an easy way to do this (except defining associated variables, which then have an artificial "type" and do not allow names containing wildcards).

Making it possible to define special variables not found in the documented code but defined only in the Doxygen documentation blocs (and allowing the standard grouping mechanisms to be used with them) would be great.

**Comment by albert-github** [comment]
A thought:
Hard coded names / strings are not the best way to have in the code (doxygen also has some), but it would be better to create variables for them and use these variables throughout the code (In C/Cpp terminology: constant character strings, defines, enums; Fortran: the `PARAMETER` attribute). When the value changes one only has to do it at one place. These variables can be used again in the normal documentation.

**Comment by YvanFournier** [comment]
Yes, I agree, and actually hinted at this workaround in the description (and use a similar in the code I work on, but this does not solve all cases:
- Such strings may be defined as `static const char[]` in C for example, with a local naming scheme, which probably should not be visible to the users, as it could confuse them with variables they cannot access;
  * making those public is of course possible, but not always preferred, for example when some keywords may change from one version of a tool to another, and we prefer that the users  dynamically test their existence rather than manage long deprecation lifecycles.
- in some cases, using common wildcards (such as simply `*`) for name patterns can be useful, and is not accessible here.
- In the code I work on (https://github.com/code-saturne/code_saturne), some filenames are reserved, and though most are accessed from C and Fortran, a few may be accessed only though higher-level Python scripts; we use Doxygen for the main code, but the general documentation and that of the scripts is based on Doxygen markdown pages, not source documentation of those scripts.
  *  To document this with the current features, our workaround is to define an "extra" C or C++ source file (which is analyzed by Doxygen but fo not appear in our Makefiles), whose name is completely artificial but still appears in the regular documentation structure.
   *  this works, but rather than have those names appear as true variables, we would prefer to simply be able to generate anchors to specific strings inside Documentation pages, and, be able to reference (and search) them with the usual mechanisms.

The last point is probably the main one. So although this is not a "blocking" feature, this type of additional option would be nice.
